src/include/migraphx/op/reverse.hpp

```diff
@@ -41,7 +41,8 @@ struct reverse
     {
         auto lens = inputs[0].lens();
         auto type = inputs[0].type();
-        return shape{type, lens};
+        shape reverse_shape{type, lens};
+        return reverse_shape.with_lens(type, lens);
```

Comment:
This can just be `return inputs[0].with_lens(inputs[0].lens())`. Maybe we could add an method like `make_packed` to `shape` class to do this, but its not necessary for this PR.

Reply:
Done, thanks!